detect_secrets/plugins/keyword.py

```diff
@@ -139,19 +139,24 @@ class KeywordDetector(BasePlugin):
 
     secret_type = 'Secret Keyword'
 
+    def __init__(self, keyword_exclude=None, **kwargs):
```

Comment:
I'm sorry I didn't explain this well, we can do

```python
    def __init__(self, keyword_exclude=None, exclude_lines_regex=None, **kwargs):
        super(KeywordDetector, self).__init__(
            exclude_lines_regex,
        )

        self.keyword_exclude = None
        if keyword_exclude:
            self.keyword_exclude = re.compile(
                keyword_exclude,
                re.IGNORECASE,
            )
```

Since `super` will call the base plugin
https://github.com/Yelp/detect-secrets/blob/50e2edf46453eacbc1e42144cf51bb0f5ce2618c/detect_secrets/plugins/base.py#L17
we'll only pass the "all plugins" `exclude_lines_regex` to it.

Reply:
np. This looks much better. :) 
So does it mean every plugin depends on this base plugin?

Reply:
Yup, that's correct, every plugin inherits from the `BasePlugin` e.g. some place will loop through plugins and call `plugin.analyze` and which is `analyze` in the base plugin that calls `analyze_string` in the base plugin that will call `analyze_string_content` in the child plugin (`KeywordDetector` in this case)  https://github.com/Yelp/detect-secrets/blob/50e2edf46453eacbc1e42144cf51bb0f5ce2618c/detect_secrets/plugins/base.py#L31-L72

Reply:
okay. That's much clear now. I am interested if there is more work in this project :)